api/envoy/config/accesslog/v3/accesslog.proto

```diff
@@ -307,7 +307,7 @@ message MetadataFilter {
   // Matcher to check metadata for specified value. For example, to match on the
   // access_log_hint metadata, set the filter to "envoy.common" and the path to
   // "access_log_hint", and the value to "true".
-  type.matcher.v3.MetadataMatcher matcher = 1;
+  type.matcher.v3.MetadataMatcher matcher = 1 [(validate.rules).message = {required: true}];
```

Comment:
https://github.com/envoyproxy/envoy/blob/master/api/API_VERSIONING.md#backwards-compatibility

> Increasing the strictness of protoc-gen-validate annotations. Exceptions may be granted for scenarios in which these stricter conditions model behavior already implied structurally or by documentation.

Assuming this change falls into this exception, because MetadataFilter can be null, @asraa ?

Reply:
It might be able to be solved in code too. `MetadataFilter` only needs an actual `MetadataMatcher matcher` so that it can have a `value`, otherwise this line crashes in code
https://github.com/envoyproxy/envoy/blob/d7c7e9a79eed80afa56a0b05cf6adf0516750d14/source/common/access_log/access_log_impl.cc#L273
because of
https://github.com/envoyproxy/envoy/blob/b6d1fecfd299d9055afb11ef7124d09e07341c68/source/common/common/matchers.cc#L34

Thinking about this more, it is possible to solve in code by only setting `value_matcher_` if there's a `matcher`, and evaluating the match by preconditioning that there's a `value_matcher_`. 
I think code may be better (at least for now)

Reply:
Opted for the code change.